src/platform/plugins/shared/dashboard/server/content_management/v1/cm_services.ts

```diff
@@ -28,68 +28,65 @@ const apiError = schema.object({
   metadata: schema.maybe(schema.object({}, { unknowns: 'allow' })),
 });
 
-const searchSourceSchema = schema.object(
+export const filterSchema = schema.object(
   {
-    type: schema.maybe(schema.string()),
-    query: schema.maybe(
+    meta: schema.object(
+      {
+        alias: schema.maybe(schema.nullable(schema.string())),
+        disabled: schema.maybe(schema.boolean()),
+        negate: schema.maybe(schema.boolean()),
+        controlledBy: schema.maybe(schema.string()),
+        group: schema.maybe(schema.string()),
+        index: schema.maybe(schema.string()),
+        isMultiIndex: schema.maybe(schema.boolean()),
+        type: schema.maybe(schema.string()),
+        key: schema.maybe(schema.string()),
+        params: schema.maybe(schema.any()),
+        field: schema.maybe(schema.string()),
+      },
+      { unknowns: 'allow' }
+    ),
+    query: schema.maybe(schema.recordOf(schema.string(), schema.any())),
+    $state: schema.maybe(
       schema.object({
-        query: schema.oneOf([
-          schema.string({
+        store: schema.oneOf(
+          [
+            schema.literal(FilterStateStore.APP_STATE),
+            schema.literal(FilterStateStore.GLOBAL_STATE),
+          ],
+          {
             meta: {
               description:
-                'A text-based query such as Kibana Query Language (KQL) or Lucene query language.',
+                "Denote whether a filter is specific to an application's context (e.g. 'appState') or whether it should be applied globally (e.g. 'globalState').",
```

Comment:
This is interesting, because any filter that is in the `global_state` store shouldn't be saved. I remember there was some talk about using standardized schemas here, but we ran into problems - is this still the idea? If so, I assume this schema is temporary?

Reply:
This is temporary until we find a better interface for filters. Likely we'll have a simpler interface in Dashboards that can write to `SearchSourceFields`.